bin/delete_pr_comment

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -eu
+
+# This script is used to delete comments on a pull request, and must be run in a buldkite PR step.
+#
+# Example:
+#   delete_pr_comment --github-token <token> --containing "part of a comment"
+
+# Check dependencies and print their versions for diagnosis purposes
+jq --version
+curl --version
+
+# Parse command line arguments
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --containing) COMMENT_CONTAINING="$2"; shift ;;
+        --github-token) GITHUB_TOKEN="$2"; shift ;;
+        *) echo "Unknown option: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+function require_option() {
+    if [[ -z "$2" ]]; then
+        echo "Error: $1 is required"
+        exit 1
+    fi
+}
+
+require_option '--containing' "$COMMENT_CONTAINING"
+require_option '--github-token' "$GITHUB_TOKEN"
```

Comment:
I tried to break this by calling without passing the option and it worked anyway, my guess is because I have `$GITHUB_TOKEN` as an export in my environment.

<img width="1280" alt="Screenshot 2023-05-03 at 1 19 57 pm" src="https://user-images.githubusercontent.com/1218433/235826798-4425b365-78aa-4f8c-b8da-472832c38275.png">

I don't think this is a blocker in the context of how we can expect this script to be used.

Reply:
Yeah, this is an intended, however undocumented, behaviour.

When invoking this script on CI, we have a few env vars defined by convention, GITHUB_TOKEN being one of them. They are used by many other tools too (`gh`, octokit etc.), it sometimes can be quite handy to just use the env var if it's not passed via CLI. 😄